src/Auth0.AspNetCore.Authentication/PushedAuthorizationRequest/PushedAuthorizationRequestHandler.cs

```diff
@@ -0,0 +1,120 @@
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Auth0.AspNetCore.Authentication.PushedAuthorizationRequest;
+
+/// <summary>
+/// Handler used to support Pushed Authorization Request with Microsoft's Open Id Connect package.
+///
+/// This is needed because there is no built-in support for PAR in ASP.NET Core.
+/// Once https://github.com/dotnet/aspnetcore/issues/51686 is implemented, we should be able to replace our
+/// internal implementation and rely on the official implementation.
+/// </summary>
+internal static class PushedAuthorizationRequestHandler
+{
+    public static async Task HandleAsync(RedirectContext context, OpenIdConnectOptions oidcOptions)
+    {
+        // Read the PAR Endpoint from the OIDC configuration.
+        var oidcConfiguration =
+            await oidcOptions.ConfigurationManager?.GetConfigurationAsync(default)!;
+        var parEndpoint = oidcConfiguration?.AdditionalData["pushed_authorization_request_endpoint"] as string;
+
+        // If PAR was enabled in the options, but no `pushed_authorization_request_endpoint` value is find
+        // in the OIDC configuration, we will skip using Pushed Authorization Request.
+        if (string.IsNullOrEmpty(parEndpoint))
+        {
+            return;
+        }
```

Comment:
@adamjmcgrath  Is no `pushed_authorization_request_endpoint` was found, but `usePushedAuthorization` was set to `true`, we will silently fallback to use regular authorization. Would that be expected or should we fail?

Reply:
The underlying OIDC SDK we use does use those endpoints:

- https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs#L402
- https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs#L876

So we already have the OpenIdConnect Configuration instance available, so I figured I'd read it for this as well.

I agree there is one place where we still use a hardcoded oauth/token, which we could eventualy remove as well.

Reply:
I updated the PR to throw an error when the OIDC config is missing.

Reply:
Looks like you're storing the configuration in memory in those other places https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs#L393-L396 - is `GetConfigurationAsync` cached somehow?

Reply:
According to the docs I find, it should cache it internally. But I will look into it.